### DISCUSSION TITLE
Add new language to Code Notes?

### DISCUSSION THREAD

**Discussion by maphew** [TriliumNext/Trilium#6051] [Ideas]
How to add a new language type to [Code Notes](https://triliumnext.github.io/Docs/Wiki/code-notes.html)?

in my case, DOS/Windows Batch or CMD syntax.

**Comment by dousha** [comment]
Trilium(Next) uses CodeMirror 5 for code notes. Adding a new language to code notes requires [adding a new language mode][1] and then adding the corresponding mode to the language list, in addition to telling CodeMirror to find your mode files. 

Unfortunately, adding a new language entry to the list might be the most easy part of this whole journey.

[1]: https://codemirror.net/5/doc/manual.html#modeapi

**Reply by maphew** [reply]
thanks for the lead! can you expand on why adding a new CodeMirror language type might be the easiest part?


what follows are jumping off points for adding a cmd language type (I'm researching from my phone)

- https://codemirror.net/examples/lang-package/
- https://github.com/codemirror/lang-example
- https://github.com/codemirror/codemirror5/issues/5497#issuecomment-1001492806
